pkg/webhooks/machine_webhook.go

```diff
@@ -1280,6 +1280,14 @@ func validateGCP(m *machinev1beta1.Machine, config *admissionConfig) (bool, []st
 		errs = append(errs, field.Invalid(field.NewPath("providerSpec", "restartPolicy"), providerSpec.RestartPolicy, fmt.Sprintf("restartPolicy must be either %s or %s.", machinev1beta1.RestartPolicyNever, machinev1beta1.RestartPolicyAlways)))
 	}
 
+	if providerSpec.ProvisioningModel != "" && providerSpec.ProvisioningModel != machinev1beta1.GCPSpotInstance {
```

Comment:
We don't actually want to allow `""`, which means this field needs to be a pointer. Only valid choices should be `nil` and `"Spot"`